dbfopen.c

```diff
@@ -1541,10 +1541,10 @@ int SHPAPI_CALL DBFWriteDateAttribute(DBFHandle psDBF, int iRecord, int iField,
         return false;
     if (lValue->day < 0 || lValue->day > 99)
         return false;
-    char dateValue[9]; /* "yyyyMMdd\0" */
-    snprintf(dateValue, sizeof(dateValue), "%04d%02d%02d", lValue->year,
-             lValue->month, lValue->day);
-    return (DBFWriteStringAttribute(psDBF, iRecord, iField, dateValue));
+    const double dateValue =
+        lValue->year * 10000 + lValue->month * 100 + lValue->day;
+    return (DBFWriteStringAttribute(psDBF, iRecord, iField,
+                                    STATIC_CAST(const char *, &dateValue)));
```

Comment:
using DBFWriteStringAttribute is super confusing. better do:
```suggestion
    double dateValue =
        lValue->year * 10000 + lValue->month * 100 + lValue->day;
    return (DBFWriteAttribute(psDBF, iRecord, iField,
                              STATIC_CAST(void *, &dateValue)));
```

Reply:
Will do and test tomorrow.

Reply:
Resolved differently by using DBFWriteAttributeDirectly. This way, an invalid date value is passed as `00000000` (see https://github.com/OSGeo/shapelib/issues/16#issuecomment-1972619646).